Content.Shared/Hands/SharedHandsSystem.cs

```diff
@@ -27,13 +27,26 @@ public override void Initialize()
 
             SubscribeLocalEvent<SharedHandsComponent, EntRemovedFromContainerMessage>(HandleContainerModified);
             SubscribeLocalEvent<SharedHandsComponent, EntInsertedIntoContainerMessage>(HandleContainerModified);
+            SubscribeLocalEvent<SharedHandsComponent, ItemPrefixChangeEvent>(OnPrefixChanged);
 
             CommandBinds.Builder
                 .Bind(ContentKeyFunctions.Drop, new PointerInputCmdHandler(DropPressed))
                 .Bind(ContentKeyFunctions.SwapHands, InputCmdHandler.FromDelegate(SwapHandsPressed, handle: false))
                 .Register<SharedHandsSystem>();
         }
 
+        private void OnPrefixChanged(EntityUid uid, SharedHandsComponent component, ItemPrefixChangeEvent args)
+        {
+            foreach (var ent in component.GetAllHeldEntities())
+            {
+                if (args.Item == ent)
+                {
+                    UpdateHandVisualizer(uid, component);
+                    return;
+                }
+            }
```

Comment:
hand code is disgusting.

Reply:
To be fair, I did this in a shitty way.

Hands have containers with container-Ids that match the hand Id. Really I should be doing a TryGetHand with a container-id, rather than getting every held item and checking if this item is one of them. So imma update this.